### DISCUSSION TITLE
Explicitly define properties which are arrays / collections.

### DISCUSSION THREAD

**Discussion by leevigraham** [schemaorg/schemaorg#3371] [Q&A]
Is there anyway to determine if a property can be an array / collection of children or a single value?

Example: https://schema.org/ItemList

The [itemListElement](https://schema.org/itemListElement) property example is an array, but the [numberOfItems](https://schema.org/numberOfItems) property is an `Integer`.

```json
{
    "@context": "https://schema.org",
    "@type": "ItemList",
    "url": "http://multivarki.ru?filters%5Bprice%5D%5BLTE%5D=39600",
    "numberOfItems": "315",
    "itemListElement": [
        {
            "@type": "Product",
            "image": "http://img01.multivarki.ru.ru/c9/f1/a5fe6642-18d0-47ad-b038-6fca20f1c923.jpeg",
            "url": "http://multivarki.ru/brand_502/",
            "name": "Brand 502",
            "offers": {
                "@type": "Offer",
                "priceCurrency": "RUB",
                "price": "4399.00"
            }
        },
        {
            "@type": "Product",
            "name": "..."
        }
    ]
}
```

**Comment by KalleOlaviNiemitalo** [comment]
If such a distinction were specified, I expect it would be a property of each Property, but schema.org itself does not define such a property.  OWL 2 defines owl:FunctionalProperty and owl:maxCardinality but those are not referenced in <https://github.com/schemaorg/schemaorg/blob/main/data/releases/22.0/schemaorg.owl>.

**Reply by mfhepp** [reply]
The OWL route does not seem feasible to me, because (1) schema.org by and large uses its own base formalism and not plain RDF-S/OWL etc. (despite some overlap) and (2) it will likely be hard to use OWL reasoning for data quality checks in a production environment.

**Comment by mfhepp** [comment]
IMO, the cleanest and most state-of-the-art way of adding this feature (aka "cardinality constraints") to schema.org would be providing a [SHACL](https://www.w3.org/TR/shacl/) shape definition that either specifies the entire schema.org vocabulary in an alternative formalism, or an isolated "schema.org data quality" SHACL shape.

For examples, see e.g. [here](https://book.validatingrdf.com/bookHtml011.html).

@danbri - maybe you could comment on plans re adding SHACL shapes for data quality rules in schema.org.
